InjectionByExample.Tests/ContainerTests.cs

```diff
@@ -0,0 +1,85 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using InjectionByExample;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace InjectionByExample.Tests
+{
+    [TestClass()]
+    public class ContainerTests
+    {
+        Container container;
+
+        public ContainerTests()
+        {
+            var containerBuilder = new ContainerBuilder();
+
+            containerBuilder.Register<Driver>();
+            containerBuilder.Register<ICar, Car>(Lifetime.InstancePerContainer);
+            containerBuilder.Register<IEngine, Engine>(Lifetime.SingleInstance);
+
+            container = containerBuilder.Build();
+        }
+
+        [TestMethod()]
+        public void ContainerWithParentNull()
+        {
+            Assert.IsNull(container.Parent);
+        }
+
+        [TestMethod()]
+        public void ContainerWithParentNotNull()
+        {
+            var childContainer = container.CreateChild();
+
+            Assert.IsNotNull(childContainer.Parent);
+        }
+
+        [TestMethod()]
+        public void ResolveWithParameterType()
+        {
+            var instance = container.Resolve(typeof(IEngine));
+
+            Assert.IsInstanceOfType(instance, typeof(Engine));
+        }
+
+        [TestMethod()]
+        public void ResolveWithGenericType()
+        {
+            var instance = container.Resolve<IEngine>();
+            Assert.IsInstanceOfType(instance, typeof(Engine));
+        }
+
+        [TestMethod()]
+        public void ResolveInstancePerContainer()
```

Comment:
A good test here would be to create 2 child containers before resolving anything. Secondly, resolve the same type with each container and then verify that they are not the same instance.

Reply:
Good sugestion. I changed it accordingly.